### PR TITLE
[t175] process generated columns on INSERT

### PR DESCRIPTION
**Pull request by seut**

### CODE REVIEW

File: sql/src/main/java/io/crate/analyze/InsertFromValuesAnalyzer.java [hunk 1]
```diff
@@ -330,4 +344,48 @@ private Object processPartitionedByValues(final ColumnIdent columnIdent, Object
         }
         return null;
     }
+
+    private Object[] processGeneratedExpressions(DocTableRelation tableRelation,
+                                             ExpressionAnalysisContext expressionAnalysisContext,
+                                             ExpressionReferenceValueAnalyzer referenceValueAnalyzer,
+                                             InsertFromValuesAnalyzedStatement context,
+                                             Object[] insertValues) {
+        referenceValueAnalyzer.values(insertValues);
```
Comment (mfelsche) [inline sql/src/main/java/io/crate/analyze/InsertFromValuesAnalyzer.java]: matter of taste, but:
couldn't you also create a new ExpressionReferenceValueAnalyzer for each call of this function?
mutable state, you know?

Comment (seut) [inline sql/src/main/java/io/crate/analyze/InsertFromValuesAnalyzer.java]: hm but there is no state at this analyzer, don't get you concerns..

File: sql/src/main/java/io/crate/analyze/expressions/ExpressionReferenceValueAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                if (reference.ident().columnIdent().equals(columnIdent)) {
+                    dataType = reference.valueType();
+                    value = values[i];
+                } else if (reference.valueType().id() == ObjectType.ID
+                           && reference.ident().columnIdent().name().equals(columnIdent.name())) {
+                    ReferenceInfo referenceInfo = tableInfo.getReferenceInfo(columnIdent);
+                    dataType = referenceInfo.type();
+                    try {
+                        int idx = 0;
+                        value = ((Map) values[i]).get(path.get(idx++));
```
Comment (mfelsche) [inline sql/src/main/java/io/crate/analyze/expressions/ExpressionReferenceValueAnalyzer.java]: what if path is null?

**Comment by mfelsche** [comment]
otherwise ack

File: sql/src/main/java/io/crate/analyze/expressions/ExpressionReferenceValueAnalyzer.java (hunk 2 above)
Comment (seut) [inline sql/src/main/java/io/crate/analyze/expressions/ExpressionReferenceValueAnalyzer.java]: good point, thx

**Pull request merged**